### ISSUE TITLE
Cache not redownloaded when deleted

### ISSUE THREAD

**Issue by gioragutt** [verdaccio/verdaccio#264]
#### Steps to reproduce:

`pm2 start vedaccio` (not exactly, I give exact path to global node modules bin file, as it doesn't work when I run `pm2 start verdaccio`, gives out node error)
`ng new test-project`
Delete `@angular` in `C:/Users/MyPc/.config/verdaccio/storage`
Delete `node_modules` in `test-project`
`npm install` inside `test-project`
`@angular` folder is not redownloaded in the cache folder, but it does appear in the `node_modules`

####  App Version:	

2.3.1

####  Config file:	
```
(#
# This is the default config file. It allows all users to do anything,
# so don't use it on production systems.
#
# Look here for more config file examples:
# https://github.com/verdaccio/verdaccio/tree/master/conf
#

# path to a directory with all packages
storage: ./storage

auth:
  htpasswd:
    file: ./htpasswd
    # Maximum amount of users allowed to register, defaults to "+inf".
    # You can set this to -1 to disable registration.
    #max_users: 1000

# a list of other known repositories we can talk to
uplinks:
  npmjs:
    url: https://registry.npmjs.org/

packages:
  '@*/*':
    # scoped packages
    access: $all
    publish: $authenticated
    proxy: npmjs

  '**':
    # allow all users (including non-authenticated users) to read and
    # publish all packages
    #
    # you can specify usernames/groupnames (depending on your auth plugin)
    # and three keywords: "$all", "$anonymous", "$authenticated"
    access: $all

    # allow all known users to publish packages
    # (anyone can register by default, remember?)
    publish: $authenticated

    # if package is not available locally, proxy requests to 'npmjs' registry
    proxy: npmjs

# log settings
logs:
  - {type: stdout, format: pretty, level: http}
  #- {type: file, path: verdaccio.log, level: info})
```

#### Additional information:

`npm get registry`: `http://localhost:4873/`

I've emptied the `storage` folder, ran `npm install -g @angular/cli`.
`@angular/cli` was indeed updated, checked by `ng -v`. But the storage folder is empty.

**Comment by juanpicado** [comment]
You miss a step here, remember all is cached locally by `npm` in `~/.npm/*` or in your case in `C:\Users\[Username]\AppData\Roaming\npm`. Those steps works for me, the fact the packages are not being re-downloaded on storage is because npm first ask your local cache, otherwise go to the registry. I had to `rm -Rf ~/.npm/*` to force `npm` to ask again for the packages.

![screen shot 2017-07-26 at 7 31 54 am](https://user-images.githubusercontent.com/558752/28605954-af8f38fe-71d4-11e7-8673-410aed915ee0.png)

I haven't tested in Windows yet, but, my first hint is `npm` that is failing you, not `verdaccio`. Please provide more info about your windows environment in order to test in a `Windows 10` I have over there.

**Comment by gioragutt** [comment]
I did have have to `npm cache clear --force`. Currently this does work for me, things are downloading as expected.

I suggest maybe adding a small RFAQ (really FAQ) which would at least note this point, alongside other relevant FAQs that should be in the README

**Comment by juanpicado** [comment]
Sure, we always can improve the documentation. Lately we have been posting new documents to improve the usage and set up, you can see https://github.com/verdaccio/verdaccio/tree/master/wiki I think your issue was merely related with `npm`, somehow, I'll point it out somewhere as a related topic because `npm` is harder than I seems so. 

`yarn` no problem, It was just suggestion.  

I'm not 100% sure what you mean with `cache fallback resolution`,  but, if you refer the local client cache.  I think until `npm` reach `verdaccio` there is nothing we can do with that.

**Comment by gioragutt** [comment]
@juanpicado regarding the `cache fallback resolution`, I do now realize there isn't much you can do about it. What I thought (at the time of writing) that maybe you have control over which cache is used (and next cache if first cache is missing what you need, and so on). But if this is done by `npm`, I guess that there's not much you can do.

**Comment by gioragutt** [comment]
@juanpicado I have a question tho.

In my company, we have an offline (not connected to `www`) `Artifactory` server, on which we run an `npm repo`. Problem is, of course, it's offline, so I can't really use it until I get the packages to the offline network.

For that I use `verdaccio`, which does the caching for me, so that I can pack it and move it to the offline network. Currently, we still don't know how to move the packages I'll move from `verdaccio` to `artifactory`, but I'm certain we'll figure it out. 

I wanted to ask if perhaps you, or someone you know, has had the same need and already did that migration, and maybe I could get some advice before we try to do it ourselves.

**Comment by juanpicado** [comment]
With `Artifactory` you mean JFrog Artifactory?

**Comment by gioragutt** [comment]
@juanpicado Yes, exactly. Specifically, the `pro` version that can host an `npm repository`

**Comment by juanpicado** [comment]
Unfortunately I have no clue about JFrog  only they 💰 for use it 😈 ., I just played around with Nexus. Perhaps others has can give you any hit about it.
btw: I think we can close this ticket, right?

**Comment by juanpicado** [comment]
They have https://github.com/JFrogDev/nexus2artifactory perhanps there is a `verdaccio/sinopia`-> `artifactory` or something like that.

**Comment by gioragutt** [comment]
@juanpicado Hey, I'm experiencing this issue once again, continuously though, even after `npm cache clean -f`.

F.e: I created a new project using `ng new`, it installed `@angular/common` 4.3.6 in the node_modules folder.

I deleted `@angular/common` from verdaccio's `storage` folder, deleted the folder from `node_modules`, `npm cache clean -f`ed again, then I ran `npm install`, and the package was not re-added to the storage.
